plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/editor/OpenDeclarationFinder.java

```diff
@@ -145,102 +145,98 @@ private void handleDescriptor(
-                    editorAgent.openEditor(
-                        file.get(),
-                        new OpenEditorCallbackImpl() {
-                          @Override
-                          public void onEditorOpened(final EditorPartPresenter editor) {
-                            Scheduler.get()
-                                .scheduleDeferred(
-                                    new Scheduler.ScheduledCommand() {
-                                      @Override
-                                      public void execute() {
-                                        if (editor instanceof TextEditor) {
-                                          ((TextEditor) editor)
-                                              .getDocument()
-                                              .setSelectedRange(
-                                                  LinearRange.createWithStart(
-                                                          descriptor.getOffset())
-                                                      .andLength(0),
-                                                  true);
-                                          editor.activate();
-                                        }
-                                      }
-                                    });
-                          }
-                        });
-                  }
-                }
-              });
+      openFileFromWorkspace(descriptor);
     }
   }
+
+  private void activateOpenedEditor(
+      OpenDeclarationDescriptor descriptor, EditorPartPresenter openedEditor) {
+    editorAgent.openEditor(
+        openedEditor.getEditorInput().getFile(),
+        new OpenEditorCallbackImpl() {
+          @Override
+          public void onEditorOpened(EditorPartPresenter editor) {
+            setCursorAndActivateEditor(editor, descriptor.getOffset());
+          }
+
+          @Override
+          public void onEditorActivated(EditorPartPresenter editor) {
+            setCursorAndActivateEditor(editor, descriptor.getOffset());
+          }
+        });
+  }
+
+  private void openFileFromWorkspace(OpenDeclarationDescriptor descriptor) {
+    appContext
+        .getWorkspaceRoot()
+        .getFile(descriptor.getPath())
+        .then(
+            file -> {
+              if (file.isPresent()) {
+                openEditor(descriptor, file.get());
+              }
+            });
+  }
+
+  private void getLibraryEntry(Path projectPath, OpenDeclarationDescriptor descriptor) {
+    ExternalLibrariesParameters entryParams =
+        dtoFactory.createDto(ExternalLibrariesParameters.class);
+    entryParams.setNodeId(descriptor.getLibId());
+    entryParams.setNodePath(descriptor.getPath());
+    entryParams.setProjectUri(projectPath.toString());
+    extensionService
+        .libraryEntry(entryParams)
+        .then(
+            entry -> {
+              getBinaryContent(projectPath, descriptor, entry);
+            });
+  }
+
+  private void getBinaryContent(
```

Comment:
A name starting with "open" is more appropriate here: "get" indicates it returns something.

Reply:
agree